src/Processors/Formats/Impl/ArrowBufferedStreams.cpp

```diff
@@ -83,6 +83,47 @@ arrow::Status RandomAccessFileFromSeekableReadBuffer::Seek(int64_t position)
     return arrow::Status::OK();
 }
 
+
+ArrowInputStream::ArrowInputStream(ReadBuffer & in_) : in(in_)
+{
+}
+
+arrow::Result<int64_t> ArrowInputStream::Read(int64_t nbytes, void * out)
+{
+    return in.read(reinterpret_cast<char *>(out), nbytes);
+}
+
+arrow::Result<std::shared_ptr<arrow::Buffer>> ArrowInputStream::Read(int64_t nbytes)
+{
+    std::string file_data;
+    {
+        WriteBufferFromString file_buffer(file_data);
+        copyData(in, file_buffer, nbytes);
```

Comment:
This could be avoided probably. This method is required to always return `nbytes` unless EOF. If needed we can probably concatenate some buffers?